### ISSUE TITLE
Repetitive rerender() causes duplication

### ISSUE THREAD

**Issue by pooledge** [testing-library/svelte-testing-library#185]
I have this very simple scenario:
```typescript
const wrapper = render(Pagination, {
	props: {
		pagination: {current: 1}
	}
});

let items = wrapper.getAllByRole('listitem');
expect(items[0].querySelector('span')?.innerHTML).toBe('1 of 20');

wrapper.rerender({
	pagination: {current: 10}
});

items = wrapper.getAllByRole('listitem');
expect(items[1].querySelector('span')?.innerHTML).toBe('10 of 20');

wrapper.rerender({
	pagination: {current: 20}
});

items = wrapper.getAllByRole('listitem');
expect(items[1].querySelector('span')?.innerHTML).toBe('20 of 20');
```

Test fails only if a `rerender()` is invoked for the 2nd time, i.e. expected `20 of 20` is actual `10 of 20`.
I'd like to `rerender()` instead of starting anew in order to ensure that certain calculations only run specified amount of times.

Does anyone also experience this?

**Comment by cornzz** [comment]
Same experience here, testing a simple Select component:

`
const optionsProp: SelectOption[] = [
        { title: 'select1', value: 'value1' },
        { title: 'select2', value: 'value2' },
        { title: 'select3', value: 'value3' }
    ]
    const component = render(Select, { props: { options: optionsProp } })
    component.debug()

    component.rerender({ props: { options: optionsProp.slice(1) } })
    component.debug()

    component.rerender({ props: { options: optionsProp.slice(2) } })
    component.debug()
`

<details>
<summary>Expected output</summary>
`
console.log
    <body>
      <div>
        <div
          class="background svelte-1bgbtbi"
        >
          <select
            class="svelte-1bgbtbi"
            id=""
            name=""
          >
            <option
              class="svelte-1bgbtbi"
              value="null"
            >
              
            </option>
            <option
              class="svelte-1bgbtbi"
              value="value1"
            >
              select1
               
            </option>
            <option
              class="svelte-1bgbtbi"
              value="value2"
            >
              select2
               
            </option>
            <option
              class="svelte-1bgbtbi"
              value="value3"
            >
              select3
               
            </option>
            
          </select>
        </div>
      </div>
    </body>

      at Object.debug (../../node_modules/@testing-library/svelte/dist/pure.js:95:40)

  console.log
    <body>
      <div>
        <div
          class="background svelte-1bgbtbi"
        >
          <select
            class="svelte-1bgbtbi"
            id=""
            name=""
          >
            <option
              class="svelte-1bgbtbi"
              value="null"
            >
              
            </option>
            <option
              class="svelte-1bgbtbi"
              value="value2"
            >
              select2
               
            </option>
            <option
              class="svelte-1bgbtbi"
              value="value3"
            >
              select3
               
            </option>
            
          </select>
        </div>
      </div>
    </body>

      at Object.debug (../../node_modules/@testing-library/svelte/dist/pure.js:95:40)

  console.log
    <body>
      <div>
        <div
          class="background svelte-1bgbtbi"
        >
          <select
            class="svelte-1bgbtbi"
            id=""
            name=""
          >
            <option
              class="svelte-1bgbtbi"
              value="null"
            >
              
            </option>
            <option
              class="svelte-1bgbtbi"
              value="value3"
            >
              select3
               
            </option>
            
          </select>
        </div>
      </div>
    </body>

      at Object.debug (../../node_modules/@testing-library/svelte/dist/pure.js:95:40)
`
</details>
<details>
<summary>Actual output</summary>
`
console.log
    <body>
      <div>
        <div
          class="background svelte-1bgbtbi"
        >
          <select
            class="svelte-1bgbtbi"
            id=""
            name=""
          >
            <option
              class="svelte-1bgbtbi"
              value="null"
            >
              
            </option>
            <option
              class="svelte-1bgbtbi"
              value="value1"
            >
              select1
               
            </option>
            <option
              class="svelte-1bgbtbi"
              value="value2"
            >
              select2
               
            </option>
            <option
              class="svelte-1bgbtbi"
              value="value3"
            >
              select3
               
            </option>
            
          </select>
        </div>
      </div>
    </body>

      at Object.debug (../../node_modules/@testing-library/svelte/dist/pure.js:95:40)

  console.log
    <body>
      <div>
        <div
          class="background svelte-1bgbtbi"
        >
          <select
            class="svelte-1bgbtbi"
            id=""
            name=""
          >
            <option
              class="svelte-1bgbtbi"
              value="null"
            >
              
            </option>
            <option
              class="svelte-1bgbtbi"
              value="value2"
            >
              select2
               
            </option>
            <option
              class="svelte-1bgbtbi"
              value="value3"
            >
              select3
               
            </option>
            
          </select>
        </div>
      </div>
    </body>

      at Object.debug (../../node_modules/@testing-library/svelte/dist/pure.js:95:40)

  console.log
    <body>
      <div>
        <div
          class="background svelte-1bgbtbi"
        >
          <select
            class="svelte-1bgbtbi"
            id=""
            name=""
          >
            <option
              class="svelte-1bgbtbi"
              value="null"
            >
              
            </option>
            <option
              class="svelte-1bgbtbi"
              value="value2"
            >
              select2
               
            </option>
            <option
              class="svelte-1bgbtbi"
              value="value3"
            >
              select3
               
            </option>
            
          </select>
        </div>
        <div
          class="background svelte-1bgbtbi"
        >
          <select
            class="svelte-1bgbtbi"
            id=""
            name=""
          >
            <option
              class="svelte-1bgbtbi"
              value="null"
            >
              
            </option>
            <option
              class="svelte-1bgbtbi"
              value="value3"
            >
              select3
               
            </option>
            
          </select>
        </div>
      </div>
    </body>

      at Object.debug (../../node_modules/@testing-library/svelte/dist/pure.js:95:40)
`
</details>

**Comment by jgbowser** [comment]
I think this issue can be closed with the merge of [#189](https://github.com/testing-library/svelte-testing-library/pull/189)

**Comment by yanick** [comment]
Agreed!
